src/pybind/mgr/cephadm/inventory.py

```diff
@@ -90,20 +90,39 @@ def is_valid_ip(ip: str) -> bool:
                 self.save()
         else:
             self._inventory = dict()
+        self._all_known_names: Dict[str, List[str]] = {}
         logger.debug('Loaded inventory %s' % self._inventory)
 
     def keys(self) -> List[str]:
         return list(self._inventory.keys())
 
     def __contains__(self, host: str) -> bool:
-        return host in self._inventory
+        return host in self._inventory or host in [hname for h in self._all_known_names.values() for hname in h]
```

Comment:
This line strikes me a bit too "code golf"-esque, especially the double comprehension.
I don't know if this is "better" but it hurts my brain less:
`itertools.chain.from_iterable(self._all_known_names.values())`

Just a thought, not a demand.